docs/StorageManagement.md

```diff
@@ -0,0 +1,235 @@
+
+- The data user should be able to leave a choice of a storage type to Fybrik and organization policies.  
+
+## Goals 
+
+- Provide modules with a connection object for writing data
+
+- Share organization storage to store temporary or persistent data while hiding the details (credentials, server URLs, service endpoints, etc.) from the data user. This means that data users will store data in organization accounts created by IT administrators.  
+
+- Govern the use of the shared storage for the given workload according to the compliance, capacity and other factors.
+
+- Optimize the shared storage for the given workload (by cost, latency to the workload)
+
+- Manage storage life cycle of the shared storage (e.g., delete temporary data after not being used, delete empty buckets, etc.) 
+
+# High Level Design
+
+## Taxonomy and structures
+
+It is crucial that the modules and the storage allocation component have a base common connection structure.  Otherwise, the components will not work together correctly when deployed.  Thus, we propose the following:
+
+- **Connection** is defined in **base** taxonomy within Fybrik repository, as it is done today:
+
+```
+// +kubebuilder:pruning:PreserveUnknownFields
+// Name of the connection to the data source
+type Connection struct {
+	// Name of the connection to the data source
+	Name                 ConnectionType   `json:"name"`
+	AdditionalProperties serde.Properties `json:"-"`
+}
+```
+
+- Fybrik defines **taxonomy layers** with schema definition for supported connections (in pkg/taxonomy/layers). Quickstart deploys Fybrik using these layers. Users can add/replace layers when deploying Fybrik.
+
+In **Phase1** we define layers for all connection types that are supported today by open-source modules, such as `s3`, `db2`, `kafka`, `arrow-flight`. (Revisit taxonomy layers used by Airbyte module.)
+
+Example of a taxonomy layer for `s3`:
+```
+  s3:
+    description: Connection information for S3 compatible object store
+    type: object
+    properties:
+      bucket:
+        type: string
+      endpoint:
+        type: string
+      object_key:
+        type: string
+      region:
+        type: string
+    required:
+    - bucket
+    - endpoint
+    - object_key
+```
+- Selection of modules is done based on connection `name`.
+
+In **Phase2** we add an optional `Category` field to `Connection`. 
+A module yaml will specify as a connection type either name or category. Optimizer will do the matching.
+
+### FybrikStorageAccount 
+Today, FybrikStorageAccount spec defines properties of s3 storage, e.g.:
+```
+spec:
+  id: theshire-object-store
+  secretRef: credentials-theshire
+  region: theshire
+  endpoint: http://s3.eu.cloud-object-storage.appdomain.cloud
+```
+
+We suggest to add `type` (connection name), `geography` and the appropriate connection properties.
```

Comment:
appropriate connection properties **taken from the taxonomy**

Reply:
OK